src/main/frontend/components/EditVenue.js

```diff
@@ -0,0 +1,232 @@
+import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router";
+import { useLocation } from "react-router-dom"
+import _ from "lodash";
+
+import StateDropdown from "./StateDropdown";
+import ErrorList from "./ErrorList";
+
+const EditVenue = (props) => {
+  let location = useLocation();
+  const [formPayload, setFormPayload] = useState({
+    name: "",
+    address: "",
+    city: "",
+    state: "",
+    zipCode: "",
+    capacity: "",
+    description: "",
+    imgUrl: "",
+    phoneNumber: ""
+  });
+
+  const [errors, setErrors] = useState({});
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [venueId, setVenueId] = useState(null);
+
+  const fetchVenue = async () => {
+    try {
+      const id = props.match.params.id;
+      setVenueId(id);
```

Comment:
While this shouldn't cause any issues we can avoid updating state at multiple levels in the same function and just use  `props.match.params.id` in the PUT fetch or define this outside of the fetch function so it is available to both functions.

Reply:
I updated to define the id outside the fetch and removed the venueId state.